cmd/podman/generate/systemd.go

```diff
@@ -97,10 +103,31 @@ func init() {
 	flags.StringVar(&format, formatFlagName, "", "Print the created units in specified format (json)")
 	_ = systemdCmd.RegisterFlagCompletionFunc(formatFlagName, common.AutocompleteFormat(nil))
 
+	flags.StringArrayVar(&wants, wantsFlagName, []string{}, "Systemd wants(network-online.target is default and couldn't be removed or rewritten)")
+	_ = systemdCmd.RegisterFlagCompletionFunc(wantsFlagName, completion.AutocompleteNone)
+
+	flags.StringArrayVar(&after, afterFlagName, []string{}, "Systemd after (network-online.target is default and couldn't be removed or rewritten)")
+	_ = systemdCmd.RegisterFlagCompletionFunc(afterFlagName, completion.AutocompleteNone)
+
+	flags.StringArrayVar(&requires, requiresFlagName, []string{}, "Systemd requires")
+	_ = systemdCmd.RegisterFlagCompletionFunc(requiresFlagName, completion.AutocompleteNone)
+
 	flags.SetNormalizeFunc(utils.TimeoutAliasFlags)
 }
 
 func systemd(cmd *cobra.Command, args []string) error {
+	if cmd.Flags().Changed(wantsFlagName) {
+		systemdOptions.Wants = &wants
```

Comment:
I feel like these should be additive (appending to the existing options), not replacements. What's the use-case for removing a dependency? We add them since they're rather important. Users removing dependencies can be easily done on generated unit files.

Reply:
No, my code may look not clear, but it doesn't remove or redefine any existing options it adds the separated section with user-defined options. The generated output will have a look such:
```
...
[Unit]
Description=Podman container-foobar.service
Documentation=man:podman-generate-systemd(1)
Wants=network-online.target
After=network-online.target
RequiresMountsFor=/var/run/containers/storage

# Manual defined dependencies
Wants=a.service b.service c.target

[Service]
...
```

Reply:
I've updated the PR according to the comments, this part of the code was removed and I hope now it looks better.